scripts/vim-patch.sh

```diff
@@ -177,19 +178,28 @@ list_vim_patches() {
   echo "  To port one of the above patches to Neovim, execute"
   echo "  this script with the patch revision as argument."
   echo
-  echo "  Examples: './scripts/vim-patch.sh 7.4.487'"
-  echo "            './scripts/vim-patch.sh 1e8ebf870720e7b671f98f22d653009826304c4f'"
+  echo "  Examples: '${BASENAME} 7.4.487'"
+  echo "            '${BASENAME} 1e8ebf870720e7b671f98f22d653009826304c4f'"
 }
 
-if [[ ${#} != 1 || "${1}" == "--help" || "${1}" == "-h" ]]; then
-  usage
+if [[ ${1} == "--help" || ${1} == "-h" ]]; then
+  usage 1
+  exit 0
+elif [[ ${#} != 1 ]]; then
+  usage 2
+  exit 1
+fi
+
+# Check if mercurial is in the user's PATH, and is executable
+if [[ ! -x $(command -v hg) ]]; then
```

Comment:
I don't know, to be honest, just noticed because it's inconsistent (then again, the regex matching apparently also doesn't need quotes). I'll try to read up on it to be able to give a better review.